source/howto/remote-access/pycharm.rst

```diff
@@ -0,0 +1,141 @@
+
+.. meta::
+    :description: How to attach a remote debugger from PyCharm to a Valohai execution
+
+.. _remote-ssh-pycharm:
+
+Debug from PyCharm
+##################
+
+.. warning::
+
+    This guide assumes you're already familiar with the fundamentals of Valohai and are working with a project that's connected to a Git repository, and configured on Valohai.
+
+.. warning::
+
+    Remote debugging is only available in PyCharm Professional.
+
+In this guide we'll look at attaching the PyCharm remote debugger to a Valohai execution.
+
+1. Sample Code
+---------------
+
+For this example we'll use the following simple example code:
+
+.. code-block:: python
+
+    import numpy as np
+    import time
+
+    for x in range(1,20) :
+        print(f"Doing computation {x}")
+        data = np.random.random((50,50))
+        sum = np.sum(data)
+        time.sleep(2)
+
+And the corresponding ``valohai.yaml`` file:
+
+.. code-block:: yaml
+
+    - step:
+      name: train
+      image: python:3.9
+      command: 
+        - pip install numpy
+        - python train.py
+
+
+1. Create a new Debug Configuration in PyCharm
+-----------------------------------------------
+
+Create a new Python Debug Server in PyCharm. The IDE host name should be ``localhost`` and the port should be the port that you want to have listening to incoming debug connections on your local machine (here 1234).
+
+.. image:: /_images/ssh_remote_debug_pycharm_conf1.png
+    :alt: Create a new Python Debug Server in PyCharm
+
+.. image:: /_images/ssh_remote_debug_pycharm_conf2.png
+    :alt: Create a new Python Debug Server in PyCharm
+
+
+You can also now start the created Debug Server and leave it to wait for the process connection. 
+
+2. Use ``pydevd_pycharm``
+-------------------------
+
+Import ``pydevd_pycharm`` and set it to wait for connection with the debugger. 
+
+.. code-block:: python
+    :emphasize-lines: 1,6-14
+
+    import pydevd_pycharm
+    import numpy as np
+    import time
+
+    # The script is halted here, until a debugger is attached
+    connected = False
+    while not connected:
+        try:
+            # The arbitrary container port 1234 on the worker machine is set to connect with the debugger.
+            # Note that this is not the same port you specified during step 1 as that is on your local machine. 
+            pydevd_pycharm.settrace('localhost', port=1234, stdoutToServer=True, stderrToServer=True)
+            connected = True
+        except Exception as e:
+            time.sleep(10)
```

Comment:
We could also
- shorten this to something like 3 seconds 
- add a print("waiting for connections")
- yeet that e as it isn't used